x-pack/plugins/security_solution/cypress/integration/network/visualization_actions.spec.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { INSPECT_MODAL } from '../../screens/inspect';
+import {
+  CREATE_CASE_FLYOUT,
+  SAVE_LENS_MODAL,
+  SELECT_CASE_MODAL,
+  VIZ_ACTIONS_BUTTONS_IN_SECURITY,
+} from '../../screens/visualization_actions';
+import { cleanKibana } from '../../tasks/common';
+
+import { closesModal } from '../../tasks/inspect';
+import { loginAndWaitForPage } from '../../tasks/login';
+import {
+  clickVizActionsAddToExistingCase,
+  clickVizActionsAddToNewCase,
+  clickVizActionsButton,
+  clickVizActionsInspect,
+  clickVizActionsOpenInLens,
+  clickVizActionsSave,
+  closeAllCasesModal,
+  closeCreateCaseFlyout,
+  closeSaveObjectModal,
+  vizActionsMenuShouldBeClosed,
+} from '../../tasks/visualization_actions';
+
+import { NETWORK_URL } from '../../urls/navigation';
+
+describe('Visualization actions', () => {
```

Comment:
This pr seems to have 0 new jest tests and only cypress tests added. Cypress tests are much slower, and so should be only used for integration level tests, not inspecting the properties of a rendered component like these tests. Testing that a handler is called after a click etc can and should be done in component unit tests

Reply:
Yes, that make sense, adding the Cypress tests for these cases because we have been testing inspect button:
`x-pack/plugins/security_solution/cypress/integration/network/inspect.spec.ts` and `x-pack/plugins/security_solution/cypress/integration/hosts/inspect.spec.ts`.

I removed some of the cases from there and think adding visualization actions' test could explain the removed scenarios.

Reply:
Just to elaborate, it's awesome you added tests, it's just until the cypress tests are run for every change in every plugin that securitySolution makes use of, they'll get inadvertently broken all the time, and block security solution prs until skipped or fixed. The plugins we make use of would likely be less against running cypress tests all the time if they ran much faster, by tweaking the config, running them in parallel, etc, but mostly just by running less of them by testing all logic possible in the faster/already run in parallel jest tests.

Reply:
I see, I'll make sure these are covered by unit tests, and talk to @MadameSheema about this, if everything is fine, I'll remove the new cypress in here.

Reply:
I agree with @kqualters-elastic, we should use cypress for those scenarios we cannot cover with unit or API tests and use it more for more user flow/E2E/integration testing.

Happy to remove scenarios that are automated in a different layer.